aws/resource_aws_db_proxy.go

```diff
@@ -0,0 +1,407 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/hashcode"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsDbProxy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsDbProxyCreate,
+		Read:   resourceAwsDbProxyRead,
+		Update: resourceAwsDbProxyUpdate,
+		Delete: resourceAwsDbProxyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateRdsIdentifier,
+			},
+			"debug_logging": {
+				Type:     schema.TypeBool,
+				Optional: true,
+			},
+			"engine_family": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					rds.EngineFamilyMysql,
```

Comment:
Can we also allow Postgres as per this announcement from today? https://aws.amazon.com/about-aws/whats-new/2020/04/amazon-rds-proxy-with-postgresql-compatibility-preview/

Reply:
Sounds good to me - hopefully the next SDK release will add new EngineFamily strings (or I'll just add "POSTGRESQL" here)

Reply:
Hey @gazoakley , `EngineFamilyPostgresql` is available already.